# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/JitSpewer.cpp
# Commit: 28c38d556fb8
# Full Hash: 28c38d556fb86f1a936d9f6a8627705230df895a
# Author: Iain Ireland <iireland@mozilla.com>
# Date: 2021-12-09 09:32:28
# Regressor Bug: 1713579
# File Overlap Count: 1
# Description:
#   Bug 1713579: Extract shape-guard folding into separate optimization r=jandem
#   
#   This patch takes the code from `MGuardShape::foldsTo` and splits it out into its own optimization. Besides jitspew, the main change is adding support for MNewPlainObject (which didn't exist when this code was first written.)
#   
#   We use `dependency()` to find the most recent store. It is possible for a node's dependency to be in a dead block. In `ValueNumberer::visitDefinition`, we check for this case; I used the same approach here.
# ==============================================================================

diff -r a09f76aa6703 -r 28c38d556fb8 js/src/jit/JitSpewer.cpp
--- a/js/src/jit/JitSpewer.cpp	Wed Dec 08 21:19:47 2021 +0000
+++ b/js/src/jit/JitSpewer.cpp	Wed Dec 08 21:19:49 2021 +0000
@@ -361,6 +361,7 @@
       "  cacheflush    Instruction Cache flushes (ARM only for now)\n"
       "  range         Range Analysis\n"
       "  wasmbce       Wasm Bounds Check Elimination\n"
+      "  shapeguards   Redundant shape guard elimination\n"
       "  logs          JSON visualization logging\n"
       "  logs-sync     Same as logs, but flushes between each pass (sync. "
       "compiled functions only).\n"
@@ -458,6 +459,8 @@
       EnableChannel(JitSpew_Pools);
     } else if (IsFlag(found, "cacheflush")) {
       EnableChannel(JitSpew_CacheFlush);
+    } else if (IsFlag(found, "shapeguards")) {
+      EnableChannel(JitSpew_RedundantShapeGuards);
     } else if (IsFlag(found, "logs")) {
       EnableIonDebugAsyncLogging();
     } else if (IsFlag(found, "logs-sync")) {